dandi/download.py

```diff
@@ -501,3 +503,70 @@ def _download_file(
         os.utime(path, (time.time(), ensure_datetime(mtime).timestamp()))
 
     yield {"status": "done"}
+
+
+class DownloadDirectory:
+    def __init__(self, filepath, filehash):
+        #: The path to which to save the file after downloading
+        self.filepath = Path(filepath)
+        #: The expected hash of the downloaded data
+        self.filehash = filehash
+        #: The working directory in which downloaded data will be temporarily
+        #: stored
+        self.dirpath = append_ext(self.filepath, ".dandidownload")
+        #: The file in `dirpath` to which data will be written as it is
+        #: received
+        self.writefile = self.dirpath / self.filepath.name
```

Comment:
I got to try it now to realize what I had missed here.  IMHO there is no need to duplicate the actual filename (which can be quite long) within `self.writefile`.  Some filesystems might even faster run into the limits of their max filepath length.  So why not to make it something like

```suggestion
        self.writefile = self.dirpath / "file"
```
?

Reply:
Done.